### DISCUSSION TITLE
Multiple ideas [bulk delete, filter, keyword mapping, override, etc.].

### DISCUSSION THREAD

**Discussion by marcuswoy** [omkarcloud/google-maps-scraper#150] [Ideas]
Hey, before I start, feel free to delete the suggestions [discussion-145](https://github.com/omkarcloud/google-maps-scraper/discussions/145) and [discussions-146](https://github.com/omkarcloud/google-maps-scraper/discussions/146), because I'll take this up again and explain it better.

**Filtering of results**

It should be possible to filter down the tasks using an input field, for example. If I am currently looking for a specific task, I have to search for it manually. If the option "Extract Cities By Country" is used, I have to search manually in more than 1000 results. If I have several queries of the type "Extract Cities By Country", it becomes even more complicated if I am interested in the task "All Task", for example.

**Delete all tasks based on a query**

It should be possible to delete a created task as a whole. Currently, each entry has to be deleted manually via the recycle bin icon, which is very inconvenient.

**Add keyword from search to "All Tasks" list**

In the "All Tasks" list, an additional column should be added (preferably at the beginning) that contains the keyword of the query against Google Maps. Example: If I use the query "Extract Cities By Country" and add the keyword coffee, the application searches for coffee in city (or something similar). This keyword should also appear in a column in the result.

**Add your own cities**

It should be possible to add your own cities for the "Extract Cities By Country" query. In addition, it would be nice if the specially added cities could be checked against the existing cities (simpler string comparison). Only cities that do not yet exist should be addedn.

**Old queries as a basis for new queries**

It would be a kick-ass feature if it were possible to restart an old query and then only get "new" results returned. So you could simply build a basic query and then use the same query a week/month later to get new results. For example, you could use the Places ID as a unique identifier.

**Comment by Chetan11-dev** [comment]
Filtering of results
Great Suggestion

Delete all tasks based on a query
I agree a checkbox based system would be Great.
Currently if you delete an All Task, All Child Task are deleted as well.

Add keyword from search to "All Tasks" list
I am not sure, if it will really be useful

Add your own cities
I am not sure, if it will really be useful

Old queries as a basis for new queries
Great Suggestion

Your suggestions put a smile on my face! I sincerely appreciate you taking the time to write them up. To be honest, my workload is quite heavy right now, so I can't commit when I'll be able to implement these features or I am may even not be able to. 
Regardless, your input is extremely extremely valuable. Thank you.
